required/latex-lab/latex-lab-math.dtx

```diff
@@ -114,11 +115,142 @@
+% \DescribeMacro\MathCollectFalse
+% \DescribeMacro\m@th
+% \DescribeMacro\SuspendTagging
+%  These commands can be used to control locally the tagging of math.
+%  The first two are new commands. |\m@th| is a standard \LaTeX\ command
+%  used in a number of places to set |\mathsurround| to zero;
+%  with active tagging it is also used to identify math
+%  that should not be tagged. Details are described below.
+%  (|\SuspendTagging| is documented in source2e.pdf.)
+%
+%  To set |\mathsurround| without disabling math tagging, use |\mathsurround\z@| directly.
+%
+%
+% \subsubsection{Using the trigger token \cs{m@th} \emph{inside} the math}
+%
+% If the grabbed math contains the token |\m@th| the tagging/processing
+% of the math is suppressed. As the math is nevertheless grabbed, this requires
+% that the end math shift token is not hidden in some other command.
+% |\m@th| sets also |\mathsurround| to zero, which is often
+% wanted in untagged math anyway.
+%
+% Text tagging is \emph{not} suppressed inside the math, e.g.,
+% |\mbox{\emph{text}}| will still create an Em-structure.
+% In contrast nested math is not tagged.
+%
+% To suppress the text tagging |\SuspendTagging{}| can be used:
+%    \begin{itemize}
+%    \item no math tagging, but |\emph| is tagged:\\
+%          |$\m@th\langle\mbox{\emph{if and only if} $x=y$}\rangle$|
+%
+%    \item no internal tagging:
+%          \\
+%         |$\m@th\SuspendTagging{}\langle\mbox{\emph{if and only if}} $x=y$}\rangle$|
+%    \end{itemize}
+%
+% The method is quite suitable for symbols and also for sub- and superscripts
+% (as long as they don't contain nested math that should be tagged).
+%
+%
+% \subsubsection{Using \cs{m@th} \emph{before} the opening \texttt{\$}}
+%
+% |\m@th| (as defined in the latex-lab-math code) sets also the internal boolean
+% which controls the grabbing  to true and so when it is used \emph{before}
+% some math it disables math grabbing and tagging for all following math
+% in the current group (including nested math).
+% Text tagging inside the math is still active, it can be disabled as above
+% with |\SuspendTagging{}|.
+%
+% When |\m@th| is used like this it is possible to reenable the math
+% tagging of nested math, by adding |\MathCollectTrue| after the opening |$|.
+%
+%    \begin{itemize}
+%    \item no math tagging, but |\emph| is tagged:\\
+%        |{\m@th$\langle\mbox{\emph{if and only if} $x=y$}\rangle$}|
+%
+%    \item  both nested math and |\emph| is tagged:\\
+%        |{\m@th$\MathCollectTrue\langle\mbox{\emph{if and only if} $x=y$}\rangle$}|
+%    \end{itemize}
+%
+% The method is suitable for symbols and sub- and superscripts too
+% (it is actually how superscripts avoid math tagging currently).
+% It can also be used in cases where the end math shift token is
+% hidden in some other command. But it is not so useful for
+% larger boxes as it sets |\mathsurround| to zero.
+% Grouping should be used to avoid side-effects on following math.
```

Comment:
wouldn't \MathCollectFalse $ \MathCollectTrue the right thing to do?

I'm not sure that documenting a mixture of \m@th and \MathCollect... is really the way to go. 

For me \m@th should only be there to catch code that is not adjusted and make it (usually) work.

Reply:
Only `\MathCollectFalse` doesn't give you the mathsurround setting. And `\m@th` outside math is used in existing code (e.g. `\textsuperscript` uses `\m@th\ensuremath`) and so I wanted to document how it behaves.

Reply:
I realized that afterwards -> therefore my later comment. However, it begs the question if the interface is quite right yet and if we need something better than this for "math-used-as-text" -- but nothing that needs to be resolved for this PR